src/sweeper/sweepers/duplicates.py

```diff
@@ -7,52 +7,54 @@
 
 import arcpy
 import re
+import os
 from xxhash import xxh64
 
-
 class DuplicateTest(object):
-    def __init__(self):
+    def __init__(self, workspace, table_name):
+        self.report = {}
+        self.workspace = workspace
+        self.table_name = table_name
 
 
-    def sweep(self, fc):
+    def sweep(self):
 
         digestDict = {}
-        duplicateDict = {}
-        dupePrintLst = []
+
         digTrim = re.compile(r'(\d+\.\d{2})(\d+)')
 
-        shps = ['Shape', 'SHAPE']
+        arcpy.env.workspace = self.workspace
 
-        fields = [f.name for f in arcpy.ListFields(fc) if f.name not in shps]
+        omitFlds = ['Shape', 'SHAPE', 'GLOBALID']
+        fields = [f.name for f in arcpy.ListFields(self.table_name) if f.name not in omitFlds]
         fields.append('SHAPE@WKT')
 
-        with arcpy.da.SearchCursor(fc, fields) as sCursor:
+        with arcpy.da.SearchCursor(self.table_name, fields) as sCursor:
             for row in sCursor:
                 shp = row[-1]
                 if shp != None:
                     coordTrim = digTrim.sub(r'\1', shp)
-                    hash = xxh64('{} {}'.format(row[1:], coordTrim))
+                    hash = xxh64('{} {}'.format(row[1:-1], coordTrim))
```

Comment:
can you explain to me what this `row[1:-1]` piece is for?

Reply:
Is this just removing the OID and SHAPE@WKT from the list of fields?

Reply:
the [1:-1] slice makes it so I hash the entire row except the objectId and shape fields. coordTrim is the shape trimmed down to 3 decimal places I think. Some of this code is borrowed from Keaton, he had a reason for trimming the shape that I can't remember now.

Reply:
i think it is to handle different precision in different geodatabases. Again I do not think it is safe to assume that the object id is in position 0 when there is no code to enforce that.